### ISSUE TITLE
Style/RegexpLiteral autocorrects `{` to unterminated literal `%r{{}`

### ISSUE THREAD

**Issue by movermeyer** [rubocop-hq/rubocop#7022]
Autocorrect for `Style/RegexpLiteral` with `EnforcedStyle: percent_r` produces invalid regex literals when the regex contains `{`  or `}` characters.

--------

## Expected behavior

Autocorrect for `Style/RegexpLiteral` with `EnforcedStyle: percent_r` would always produce valid regex literals.

ie. `MY_REGEX = /{/` gets changed to `MY_REGEX = %r{\{}`
ie. `MY_REGEX = /}/` gets changed to `MY_REGEX = %r{\}}`

## Actual behavior

`MY_REGEX = /{/` gets changed to `MY_REGEX = %r{{}`, which isn't a valid regex literal (it isn't closed since the `{` isn't escaped).

## Steps to reproduce the problem

**my_file.rb:**
```ruby
MY_REGEX = /{/
```

**.rubocop.yml:**
```yaml
Style/RegexpLiteral:
  EnforcedStyle: percent_r
```

```bash
rubocop --only 'Style/RegexpLiteral' -a my_file.rb
...
Offenses:

my_file.rb:1:12: F: Lint/Syntax: unterminated string meets end of file
(Using Ruby 2.5 parser; configure using TargetRubyVersion parameter, under AllCops)
MY_REGEX = %r{{}
           ^
my_file.rb:1:12: C: [Corrected] Style/RegexpLiteral: Use %r around regular expression.
MY_REGEX = /{/
           ^^^

1 file inspected, 2 offenses detected, 1 offense corrected
```

## RuboCop version

```
rubocop -V
0.68.1 (using Parser 2.6.3.0, running on ruby 2.5.3 x86_64-darwin18)
```
